### PR TITLE
WIP Docs/how to create custom column map expectations WIP

### PR DESCRIPTION
**Pull request by abegong**
Changes proposed in this pull request:
- Adds `creating_custom_expectations/how_to_create_custom_column_map_expectations.md `, with small supporting changes in other docs.
-

I'm creating this PR to get a jump on any test failures.

### CODE REVIEW

File: docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_map_expectations.md [hunk 1]
```diff
@@ -1,31 +1,249 @@
...
+```
+
+When in doubt, the next step to implement is the first one that doesn't have a ✔ next to it. This guide covers the first four steps on the checklist.
+
+#### 4. Change the Expectation class name and add a docstring
+
+Let's start by updating your Expectations's name and docstring.
+
+Replace the Expectation class name
+```python
```
Comment (donaldheppner) [inline docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_map_expectations.md]: Does this code exist in the repo? Can it be referenced here and put under test? Example from "how_to_configure_a_configuredassetdataconnector.md"
<pre>
```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_configure_a_configuredassetdataconnector.py#L1-L4
```
</pre>

**Review by donaldheppner — CHANGES_REQUESTED** [review]

**Comment by abegong** [comment]
This doc is ready for close review of the text.

It still needs to be brought under test. That will depend on a separate PR to refactor and add types to the report object generates by `Expectation.run_diagnostics` and `Expectation.generate_diagnostic_checklist`.

File: docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_map_expectations.md (hunk 1 above)
Comment (austiezr) [reply docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_map_expectations.md]: @donaldheppner This code now exists in this pull, references have been updated as above.

**Review by Rachel-Reverie — COMMENTED** [review]
It looks good to me.  I'd like to double check the documentation changes as they'll appear to users, but it looks like the netlify build failed on a broken link.  I'll give it a second look when that is resolved.

**Review by Rachel-Reverie — APPROVED** [review]

**Pull request merged**